js/components/app.js

```diff
@@ -46,6 +46,25 @@ const DEFAULT_AUTHORED_STATE = {
   chartAvgs: false
 };
 
+// If the state is valid return it, otherwise return undefined
+function validAuthoredState(state) {
+  return state && state.columns ? state : undefined;
+}
+
+function objectOrJSONParse(value) {
+  if (typeof value === 'string') {
+    // In some cases an empty string is passed intead of valid JSON
+    // or an object
+    if (value === "") {
+      return null;
+    } else {
+      return JSON.parse(value);
```

Comment:
Do you need to worry about the case of a non-parseable string? If so, you could wrap in a try/catch block.

Reply:
Yeah it would be good to do that, but I'll punt it for now though. If there is an error in the string we should see the exception in the console and hopefully can track that down.  Ideally we'd put up some user dialog so if this happens a user knows something is awry.